orchestrator/migrations/versions/schema/2023-03-06_e05bb1967eff_add_subscriptions_search_view.py

```diff
@@ -0,0 +1,156 @@
+"""Add Subscriptions search view.
+
+Revision ID: e05bb1967eff
+Revises: bed6bc0b197a
+Create Date: 2023-03-06 10:09:55.675588
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "e05bb1967eff"
+down_revision = "bed6bc0b197a"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    conn = op.get_bind()
+    subscriptions_search_query = """
+WITH rt_info AS (SELECT s.subscription_id,
+                        concat_ws(', ',
+                                  string_agg(rt.resource_type || ':' || siv.value, ', ' ORDER BY rt.resource_type),
+                                  string_agg(distinct 'subscription_instance_id' || ':' || si.subscription_instance_id, ', ')
+                            ) AS rt_info
+                 FROM subscription_instance_values siv
+                          JOIN resource_types rt ON siv.resource_type_id = rt.resource_type_id
+                          JOIN subscription_instances si ON siv.subscription_instance_id = si.subscription_instance_id
+                          JOIN subscriptions s ON si.subscription_id = s.subscription_id
+                 GROUP BY s.subscription_id),
+     sub_prod_info AS (SELECT s.subscription_id,
+                              array_to_string(
+                                      ARRAY ['subscription_id:' || s.subscription_id,
+                                          'status:' || s.status,
+                                          'insync:' || s.insync,
+                                          'subscription_description:' || s.description,
+                                          'note:' || coalesce(s.note, ''),
+                                          'customer_id:' || s.customer_id,
+                                          'product_id:' || s.product_id],
+                                      ', ') AS sub_info,
+                              array_to_string(
+                                      ARRAY ['product_name:' || p.name,
+                                          'product_description:' || p.description,
+                                          'tag:' || p.tag,
+                                          'product_type:', p.product_type],
+                                      ', ') AS prod_info
+                       FROM subscriptions s
+                                JOIN products p ON s.product_id = p.product_id),
+     fi_info AS (SELECT s.subscription_id,
+                        string_agg(fi.name || ':' || fi.value, ', ' ORDER BY fi.name) AS fi_info
+                 FROM subscriptions s
+                          JOIN products p ON s.product_id = p.product_id
+                          JOIN fixed_inputs fi ON p.product_id = fi.product_id
+                 GROUP BY s.subscription_id),
+     cust_info AS (SELECT s.subscription_id,
+                          string_agg('customer_description: ' || scd.description, ', ') AS cust_info
```

Comment:
Whitespace here at the end `'customer_description: '`
Not sure if this matters but didn't see it anywhere else in the query

Reply:
That's the separator between the joined string. Otherwise it would be `item1,item2` instead of `item1, item2`. I guess it doesn't matter after parsing, because comma's and whitespace are delimiters anyway.